controllers/subnamespace_controller.go

```diff
@@ -58,52 +61,79 @@ func (r *SubnamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	// Hierarchy 'breadcrumb' name, built from ownerNamespace and subspace.
-	childNamespaceName = ownerNamespace.Name + "-" + subspace.Name
+	childNamespaceName := GetChildNamespaceName(&ownerNamespace, &subspace)
 
 	// current subspace's phase flow
 	switch currentPhase := subspace.Status.Phase; currentPhase {
 	case danav1alpha1.None:
-		if err := r.InitializeSubspace(ctx, ownerNamespace, subspace, childNamespaceName); err != nil {
-			log.Error(err, "Could not update Subspace Phase from 'None' to 'Missing'")
 
+		if err := r.InitSubspace(ctx, log, &ownerNamespace, &subspace, childNamespaceName); err != nil {
+			return ctrl.Result{}, err
 		}
+		log.Info("Subspace inited")
+		return ctrl.Result{}, nil
+
 	case danav1alpha1.Missing:
-		if err := r.Get(ctx, client.ObjectKey{Namespace: "", Name: childNamespaceName}, &childNamespace); err != nil {
-			if errors.IsNotFound(err) {
-				// Child Namespace is not found, therefore we are creating one
-				if requeueResult, err := r.CreateNamespace(ctx, childNamespace, ownerNamespace, subspace); err != nil {
-					log.Error(err, "Could not create Namespace ")
-					return ctrl.Result{}, err
-				} else {
-					return ctrl.Result{Requeue: requeueResult}, nil
-				}
+
+		shouldCreateChildNs, err := r.IsChildNsMissing(ctx, log, childNamespaceName)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if shouldCreateChildNs {
+			if err := r.CreateNamespace(ctx, log, &ownerNamespace, &subspace); err != nil {
+				return ctrl.Result{}, err
 			}
+			log.Info("child namespace created")
+			return ctrl.Result{Requeue: true}, nil
 		} else {
```

Comment:
Could have combine the conditions with using the r.IsChildNsMissing as a condition